### ISSUE TITLE
[MEAR-287] Failed to create target directory when run without clean

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-ear-plugin#456]
**[Mikhail Gavrilov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zzzlobster)** opened **[MEAR-287](https://issues.apache.org/jira/browse/MEAR-287?redirect=false)** and commented

Steps to reproduce:
1. Checkout https://github.com/mabrarov/dockerfile-test
2. Set version of maven-ear-plugin in pom.xml to '3.1.0'
3. Execute: `mvn clean package -Ddocker.skip && mvn package -Ddocker.skip`

```bash
$ git clone --branch develop https://github.com/mabrarov/dockerfile-test.git && \
 mvn -f dockerfile-test/pom.xml package -Dmaven-ear-plugin.version=3.1.0 -Ddocker.skip && \
 mvn -f dockerfile-test/pom.xml package -Dmaven-ear-plugin.version=3.1.0 -Ddocker.skip
...
[INFO] --- maven-ear-plugin:3.1.0:ear (default-ear) @ ear ---
[INFO] Copying artifact [war:org.mabrarov.dockerfile-test:war:1.1.7-SNAPSHOT] to [org.mabrarov.dockerfile-test-war-1.1.7-SNAPSHOT.war]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for dockerfile-test 1.1.7-SNAPSHOT:
[INFO] 
[INFO] dockerfile-test .................................... SUCCESS [  1.728 s]
[INFO] war ................................................ SUCCESS [  3.652 s]
[INFO] ear ................................................ FAILURE [  0.371 s]
[INFO] base-image ......................................... SUCCESS [  2.297 s]
[INFO] hollow-image ....................................... SUCCESS [  1.152 s]
[INFO] app-image .......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.011 s (Wall Clock)
[INFO] Finished at: 2020-10-01T14:34:15+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-ear-plugin:3.1.0:ear (default-ear) on project ear: Failed to create directory XXX...\dockerfile-test\ear\target\temp\org.mabrarov.dockerfile-test-war-1.1.7-SNAPSHOT.war -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :ear

```

The same is described [here](https://github.com/apache/maven-ear-plugin/pull/15#issuecomment-702158521)

---

**Affects:** 3.1.0

**Issue Links:**
- [MEAR-285](https://issues.apache.org/jira/browse/MEAR-285) EarMojo fails to handle assorted IO Errors
 (_**"is caused by"**_)

**Remote Links:**
- [GitHub Pull Request #17
](https://github.com/apache/maven-ear-plugin/pull/17)
- [Test for the fix provided in pull request #17
](https://github.com/zzzLobster/maven-ear-plugin/pull/1)

1 votes, 3 watchers

**Comment by jira-importer** [comment]
**[Mikhail Gavrilov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zzzlobster)** commented

https://github.com/apache/maven-ear-plugin/pull/17

**Comment by jira-importer** [comment]
**[Marat Abrarov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=abrarovm)** commented

Integration test for the fix provided in [pull request #17](https://github.com/apache/maven-ear-plugin/pull/17) can be found in https://github.com/zzzLobster/maven-ear-plugin/pull/1
